### PR TITLE
Master update terraform golang

### PR DESCRIPTION
**Pull request by jhixson74**
Update terraform go modules to use v1.20.

https://issues.redhat.com/browse/CORS-2695

### CODE REVIEW

File: terraform/Makefile [hunk 1]
```diff
@@ -25,7 +23,7 @@ go-mod-tidy-vendor: $(GO_MOD_TIDY_TARGETS)
 # The -compat=1.18 is needed for the openstack provider. The provider uses golang.org/x/mod, which go 1.18 selects
 # as v0.3.0 but go 1.16 selects as v0.4.2.
 $(GO_MOD_TIDY_TARGETS): go-mod-tidy-vendor.%:
-	cd providers/$* && go mod tidy -compat=1.18 && go mod vendor
+	cd providers/$* && go mod edit -go=$(GOVERSION) && go mod tidy -compat=1.18 && go mod vendor
```
Comment (r4f4) [inline terraform/Makefile:26]: Should we also bump the compat version?
```suggestion
	cd providers/$* && go mod edit -go=$(GOVERSION) && go mod tidy -compat=$(GOVERSION) && go mod vendor
```

Comment (jhixson74) [reply terraform/Makefile:26]: No idea. I was going to ask the guys that did this if it should be bumped. I will find out. I am currently not familiar with what this actually does.

**Comment by sadasu** [comment]
/override ci/prow/e2e-alibaba

Hasn't passed since April this year.

**Comment by patrickdillon** [comment]
It's looking more and more likely that we will be trying to remove TF as quickly as possible. Considering that, would it still make sense to add the Makefile?

**Comment by patrickdillon** [comment]
The top-level makefile warrants its own PR, because the Go version update should be a quick merge, but the makefile should probably be reviewed more carefully by the team.

**Comment by sadasu** [comment]
@jhixson74 this work may have been completed as part of https://github.com/openshift/installer/pull/7590. Could you PTAL?

**Pull request closed without merging**